dms_field/models/dms_field_template.py

```diff
@@ -103,24 +103,43 @@ def _copy_files_from_directory(self, directory, new_directory):
         for file in directory.file_ids:
             file.copy({"directory_id": new_directory.id})
 
-    def _set_groups_from_directory(self, directory, record):
-        groups = self.env["dms.access.group"]
-        for group in directory.group_ids:
-            group_name = _("Autogenerate group from %(model)s (%(name)s) #%(id)s") % {
-                "model": record._description,
-                "name": record.display_name,
-                "id": record.id,
-            }
-            new_group = group.copy({"name": group_name, "directory_ids": False})
-            # Apply sudo() because the user may not have permissions to access
-            # ir.model.fields.
-            user_field = self.sudo().user_field_id
-            if user_field:
-                user = record[user_field.name]
-                if user:
-                    new_group.write({"explicit_user_ids": [(4, user.id)]})
-            groups += new_group
-        return groups
+    def _prepare_autogenerated_group(self, record):
+        group_name = _("Autogenerated group from %(model)s (%(name)s) #%(id)s") % {
+            "model": record._description,
+            "name": record.display_name,
+            "id": record.id,
+        }
+        vals = {
+            "name": group_name,
+            # We need to set all the permissions so that the user canmanage his
+            # documents (directories and files)
+            "perm_create": True,
+            "perm_write": True,
+            "perm_unlink": True,
+            "dms_field_ref": "%s,%s" % (record._name, record.id),
+        }
+        # Apply sudo() because the user may not have permissions to access
+        # ir.model.fields.
+        user_field = self.sudo().user_field_id
+        if user_field:
+            user = record[user_field.name]
+            if user:
+                vals.update({"explicit_user_ids": [(4, user.id)]})
+        return vals
```

Comment:
The rest of the possible groups are not inherited here. Why?

Reply:
If you look at the `_prepare_directory_vals()` method you will see that the groups in the directory to be created are:
- Template folder groups (`directory.group_ids`)
- Auto-generated group (with full permissions) and the corresponding explicit user.